packages/babel-plugin-proposal-optional-chaining/src/index.js

```diff
@@ -34,6 +34,11 @@ export default declare((api, options) => {
             optionalPath.node.type = "CallExpression";
             optionalPath = optionalPath.get("callee");
           }
+
+          // unwrap a TSNonNullExpression if need
+          if (optionalPath.isTSNonNullExpression()) {
+            optionalPath.replaceWith(optionalPath.node.expression);
```

Comment:
I'm not sure that this will correctly fix the bug. We may be able to see it more in test cases like `a?.b!.c.d` and `a?.b.c!.d`